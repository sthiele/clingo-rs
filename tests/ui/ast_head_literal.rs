use clingo::*;

fn main() {
    let sym = Symbol::create_id("test", true).unwrap();
    let term = ast::Term::from(sym);
    let mut lit = ast::Literal::from_term(ast::Sign::NoSign, &term);
    let hlit = ast::HeadLiteral::from(&lit);
    lit = ast::Literal::from_term(ast::Sign::NoSign, &term);
    let _end = (lit, hlit);
}