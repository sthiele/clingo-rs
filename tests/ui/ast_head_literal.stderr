error[E0506]: cannot assign to `lit` because it is borrowed
 --> $DIR/ast_head_literal.rs:8:5
  |
7 |     let hlit = ast::HeadLiteral::from(&lit);
  |                                       ---- borrow of `lit` occurs here
8 |     lit = ast::Literal::from_term(ast::Sign::NoSign, &term);
  |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ assignment to borrowed `lit` occurs here
9 |     let _end = (lit, hlit);
  |                      ---- borrow later used here
